File: src/main/java/cn/pprocket/items/Topic.java
package cn.pprocket.items;

import cn.pprocket.HeyClient;
import cn.pprocket.utils.ParamsBuilder;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.SneakyThrows;
import okhttp3.Request;
import org.example.cn.pprocket.utils.app.AppParamsBuilder;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Topic {
    private String name;
    private int id;
    private String icon;
    public static Topic DAILY = new Topic("盒友杂谈", 7214,"");
    public static Topic LOVE = new Topic("情投一盒", 416158,"");
    public static Topic SCHOOL = new Topic("校园生活", 549999,"");
    public static Topic HARDWARE = new Topic("数码硬件", 18745,"");
    public static Topic WORK = new Topic("职场工作", 550000,"");
    public static Topic HOTS = new Topic("热点", -1,"");
    public static Topic RECOMMEND = new Topic("推荐", -2,"");
    public static Topic MAX = new Topic("Max家", 475512,"");

    @SneakyThrows
    public static List<Topic> getTopics() {
        Map<String, String> params = new HashMap<String, String>();
        params.put("type","list");
        params.put("is_post","1");
        params.put("post_tab","1");
        String url = "https://api.xiaoheihe.cn/bbs/app/api/topic/index?" + new ParamsBuilder(params).build("/bbs/app/api/topic/index/");
        String string = HeyClient.INSTANCE.get(url);
        List<Topic> topics = new ArrayList<>();
        JsonParser.parseString(string).getAsJsonObject().getAsJsonArray("topics_list").forEach(e -> {
            e.getAsJsonObject().getAsJsonArray("children").forEach( e0 -> {
                JsonObject e1 = e0.getAsJsonObject();

                Topic topic = new Topic();
                topic.setId(e1.get("topic_id").getAsInt());
                topic.setName(e1.get("name").getAsString());
                topic.setIcon(e1.get("pic_url").getAsString());
                topics.add(topic);
            });
        });
        return topics;
    }
}

File: src/commonMain/kotlin/cn/pprocket/items/Topic.kt
package cn.pprocket.items

import cn.pprocket.HeyClient.domain
import cn.pprocket.HeyClient.get
import cn.pprocket.utils.ParamsBuilder
import kotlinx.serialization.json.*



data class Topic(
    val name: String,
    val id: Int,
    val icon: String,
) {


    companion object {
        var DAILY: Topic = Topic("盒友杂谈", 7214, "")
        var LOVE: Topic = Topic("情投一盒", 416158, "")
        var SCHOOL: Topic = Topic("校园生活", 549999, "")
        var HARDWARE: Topic = Topic("数码硬件", 18745, "")
        var WORK: Topic = Topic("职场工作", 550000, "")
        var HOTS: Topic = Topic("热点", -1, "")
        var RECOMMEND: Topic = Topic("推荐", -2, "")
        var MAX: Topic = Topic("Max家", 475512, "")


        suspend fun fetchTopics(): List<Topic> {
            val params: MutableMap<String, String> = HashMap()
            params["type"] = "list"
            params["is_post"] = "1"
            params["post_tab"] = "1"
            val url =
                "${domain}/bbs/app/api/topic/index?" + ParamsBuilder(params).build("/bbs/app/api/topic/index/")
            val string = get(url)
            val topics = mutableListOf<Topic>()
            Json.decodeFromString<JsonObject>(string).get("topics_list")!!.jsonArray.forEach {
                it.jsonObject.get("children")!!.jsonArray.forEach {
                    val o = it.jsonObject
                    val topic = Topic(
                        id = o.get("topic_id")!!.jsonPrimitive.int,
                        name = o.get("name")!!.jsonPrimitive.content,
                        icon = o.get("pic_url")!!.jsonPrimitive.content
                    )
                    topics.add(topic)
                }
            }
            return topics
        }
    }
}

